src/IIIFPresentation/API.Tests/Converters/ManifestConverterTests.cs

```diff
@@ -143,5 +143,43 @@ public void SetGeneratedFields_SetsParentAndSlug_FromHierarchyUsingFactory()
         // Assert
         result.Slug.Should().Be("other-slug");
         result.Parent.Should().Be("http://base/0/collections/other-parent", "Always use FlatId");
-    } 
+    }
+    
+    [Fact]
+    public void SetGeneratedFields_SetsCanvasPainting_IfPresent()
+    {
+        // Arrange
+        var iiifManifest = new PresentationManifest();
+        var dbManifest = new DBManifest
+        {
+            CustomerId = 123,
+            Created = DateTime.UtcNow,
+            Modified = DateTime.UtcNow.AddDays(1),
+            CreatedBy = "creator",
+            ModifiedBy = "modifier",
+            Id = "id",
+            Hierarchy = [new Hierarchy { Slug = "slug" }],
+            CanvasPaintings =
+            [
+                new CanvasPainting
+                {
+                    CanvasOriginalId = new Uri("http://example.test/canvas1"),
+                    CustomerId = 123,
+                    Id = "the-canvas",
+                    ChoiceOrder = 10,
+                    CanvasOrder = 100
+                }
+            ]
+        };
+        
+        // Act
+        var result = iiifManifest.SetGeneratedFields(dbManifest, pathGenerator);
+
+        // Assert
+        var cp = result.PaintedResources.Single().CanvasPainting;

```

Comment:
Would it be worth changing `cp` to something more descriptive like `itemCanvasPainting`? (This probably doesn't matter too much, but thought I'd add as it seems like shorthand variable names like these are usually reserved for loops and lambda expressions)

Reply:
Feels okay in the context of a single test?